charts/datadog/templates/cluster-agent-rbac.yaml

```diff
@@ -29,6 +29,16 @@ rules:
   - list
   - watch
   - create
+{{- if .Values.datadog.kubernetesUseEndpointSlices }}
+- apiGroups:
+  - "discovery.k8s.io"
+  resources:
+  - endpointslices
+  verbs:
+  - get
+  - list
+  - watch
+{{- end }}
```

Comment:
IMO it can be part of the default RBAC, because we already have it for service and endpoints